packages/components/src/components/SquigglePlayground.tsx

```diff
@@ -420,10 +304,137 @@ export const SquigglePlayground: FC<PlaygroundProps> = ({
       </HeadedSection>
     </div>
   );
+};
+
+const PlayControls: React.FC<{
+  autoplay: boolean;
+  playing: boolean;
+  stale: boolean;
+  onAutoplayChange: (value: boolean) => void;
+  play: () => void;
+}> = ({ autoplay, playing, stale, onAutoplayChange, play }) => {
+  const CurrentPlayIcon = playing ? RefreshIcon : PlayIcon;
+
+  return (
+    <div className="flex space-x-1 items-center">
+      {autoplay ? null : (
+        <button onClick={play}>
+          <CurrentPlayIcon
+            className={clsx(
+              "w-8 h-8",
+              playing && "animate-spin",
+              stale ? "text-indigo-500" : "text-gray-400"
+            )}
+          />
+        </button>
+      )}
+      <Toggle
+        texts={["Autoplay", "Paused"]}
+        icons={[CheckCircleIcon, PauseIcon]}
+        status={autoplay}
+        onChange={onAutoplayChange}
+      />
+    </div>
+  );
+};
+
+export const SquigglePlayground: FC<PlaygroundProps> = ({
+  initialSquiggleString = "",
+  height = 500,
+  showTypes = false,
+  showControls = false,
+  showSummary = false,
+  logX = false,
+  expY = false,
+  code: controlledCode,
+  onCodeChange,
+  onSettingsChange,
+  showEditor = true,
+}) => {
+  const [uncontrolledCode, setUncontrolledCode] = useState(
+    initialSquiggleString
+  );
+  const [renderedCode, setRenderedCode] = useState(""); // used only if autoplay is false
+  const code = controlledCode ?? uncontrolledCode;
+
+  const [imports, setImports] = useState({});
+
+  const {
+    register,
+    control,
+    setValue: setFormValue,
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      sampleCount: 1000,
+      xyPointLength: 1000,
+      chartHeight: 150,
+      showTypes,
+      showControls,
+      logX,
+      expY,
+      showSummary,
+      showEditor,
+      leftSizePercent: 50,
+      showSettingsPage: false,
+      diagramStart: 0,
+      diagramStop: 10,
+      diagramCount: 20,
+      autoplay: true,
+    },
+  });
+  const vars = useWatch({
+    control,
+  });
+
+  useEffect(() => {
+    onSettingsChange?.(vars);
+  }, [vars, onSettingsChange]);
+
+  const chartSettings = useMemo(
+    () => ({
+      start: Number(vars.diagramStart),
+      stop: Number(vars.diagramStop),
+      count: Number(vars.diagramCount),
+    }),
+    [vars.diagramStart, vars.diagramStop, vars.diagramCount]
+  );
+
+  const env: environment = useMemo(
+    () => ({
+      sampleCount: Number(vars.sampleCount),
+      xyPointLength: Number(vars.xyPointLength),
+    }),
+    [vars.sampleCount, vars.xyPointLength]
+  );
+
+  const [playing, setPlaying] = useState(false); // used in manual play mode only
```

Comment:
maybe rename to something like, ``isAutoRunning`` and ``setIsAutoRunning``

Reply:
Also, I'd flag that we could probably use a reducer or something else a bit more sophisticated for the Squiggle Playground. All of this state is a bit much.